pkg/build/builder/cmd/builder.go

```diff
@@ -65,8 +65,8 @@ func run(builderFactory factoryFunc, scmAuths []scmauth.SCMAuth) {
 			dockercfg.PullAuthType,
 		)
 	}
-	if len(build.Parameters.Source.SourceSecretName) > 0 {
-		if err := setupSourceSecret(build.Parameters.Source.SourceSecretName, scmAuths); err != nil {
+	if build.Parameters.Source.SourceSecret != nil {
```

Comment:
This code assumes that if the `SourceSecret` is non-nil, then it will always have a `Name`.  That is guaranteed.  Add validation code to ensure that this is the case.